File: Data Structures & Algorithms/insert-into-a-binary-search-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        TreeNode root1 = root;
        if (root1 == null) {
            return new TreeNode(val);
        }
        if (root1.val <= val) {
            root1.right = insertIntoBST(root1.right, val);
        }
        else if (root1.val > val) {
            root1.left = insertIntoBST(root1.left, val);
        }
    return root1;
    }
}

File: Data Structures & Algorithms/house-robber-iii/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rob(self, root: Optional[TreeNode]) -> int:
        def check(root):
            if root is None:
                return (0, 0)

            

            left_rob, left_not_rob = check(root.left)
            right_rob, right_not_rob = check(root.right)

            rob_val = left_not_rob + right_not_rob + root.val
            not_rob_val = max(left_rob, left_not_rob) + max(right_rob, right_not_rob)

            return (rob_val, not_rob_val)
        
        return max(check(root))
        



        